### PR TITLE
implementing clamav scan

### PR DESCRIPTION
**Pull request by SoLetsDev**

### CODE REVIEW

File: backend/src/components/fileUtils.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    if (clamAVScanResult.includes('FOUND')) {
+      log.error('ClamAV scan found possible virus')
+      return false
+    }
+  } catch (e) {
+    // if virus scan is not to be performed/cannot be performed
+    log.error('ClamAV Scanner was not found: ' + e)
+    //assumption if scanner is not available we allow files to be uploaded.
+  }
+  log.info('ClamAV scan found no virus in file, allowing upload')
```
Comment (weskubo-cgi) [inline backend/src/components/fileUtils.js:41]: This log statement will execute even if ClamAV is down. Move to end of try block.

File: backend/src/components/fileUtils.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  // no virus found in file
+  next()
+}
+
+async function scanFile(buffer) {
+  try {
+    const ClamAVScanner = createScanner(config.get('clamav:host'), Number(config.get('clamav:port')))
+    const clamAVScanResult = await ClamAVScanner.scanBuffer(buffer, 3000, 1024 * 1024)
+    if (clamAVScanResult.includes('FOUND')) {
```
Comment (weskubo-cgi) [inline backend/src/components/fileUtils.js:32]: I'd be inclined to make this a const. E.g. RESULT_FOUND

File: frontend/src/components/messages/ReplyRequestDialog.vue [hunk 3]
```diff
@@ -109,13 +111,17 @@ export default {
...
-          await this.createReply(this.assistanceRequestId)
           await DocumentService.createDocuments(this.uploadedDocuments, this.assistanceRequestId)
+          await this.createReply(this.assistanceRequestId)
           await this.updateStatusToAssigned(this.assistanceRequestId)
           await this.updateStoredAssistanceRequest(this.assistanceRequestId)
           this.$emit('reply-success-event', true) // emit success to flag showing success message
         } catch (error) {
-          console.log(`Submit processing for reply failed  - ${error}`)
+          if (error?.response?.data?.status === 422) {
+            this.setFailureAlert('Submit processing for reply failed supporting documents failed virus scan', error)
```
Comment (weskubo-cgi) [inline frontend/src/components/messages/ReplyRequestDialog.vue:121]: This is a little verbose for the user. Maybe something like "Supporting documents not saved. Virus scan failed."

File: frontend/src/views/applications/OperatingCostsView.vue [hunk 4]
```diff
@@ -190,7 +190,11 @@ export default {
           this.setSuccessAlert('Application saved successfully')
         }
       } catch (error) {
-        this.setFailureAlert('Failed to save your application', error)
+        if (error?.response?.data?.status === 422) {
+          this.setFailureAlert('Failed to save your application supporting documents failed virus scan', error)
```
Comment (weskubo-cgi) [inline frontend/src/views/applications/OperatingCostsView.vue:194]: This is a little verbose for the user. Maybe something like "Supporting documents not saved. Virus scan failed."

**Review by weskubo-cgi — CHANGES_REQUESTED** [review]

File: backend/src/components/fileUtils.js (hunk 1 above)
Comment (SoLetsDev) [reply backend/src/components/fileUtils.js]: moved it

File: backend/src/components/fileUtils.js (hunk 2 above)
Comment (SoLetsDev) [reply backend/src/components/fileUtils.js]: changed to a constant VIRUS_FOUND

**Review by weskubo-cgi — APPROVED** [review]

**Pull request merged**